### PR TITLE
Optimizations 1

### PR DESCRIPTION
**Pull request by radex**
- Add function name for observer for easier debugging
- Hoist most of the class implementation to root + some microoptimizatins for performaaance

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -50,6 +65,155 @@ function getTriggeringProps<PropsInput: {}>(
...
+    this.triggerProps = triggerProps
+    this.getNewProps = getNewProps
+    this.state = {
+      isFetching: true,
+      values: {},
+      triggeredFromProps: getTriggeringProps(props, triggerProps),
+    }
+
+    // The recommended React practice is to subscribe to async sources on `didMount`
+    // Unfortunately, that's slow, because we have an unnecessary empty render even if we
```
Comment (kokusGr) [inline src/index.js]: Hmm I am thinking actually how slow is this empty render and subscribing on mount? It adds a lot of complexity and as you stated might not by compatible with future React changes.

**Comment by kokusGr** [comment]
@radex Oh, I actually cannot merge in this repo 😛

File: src/index.js (hunk 1 above)
Comment (radex) [reply src/index.js]: This part should work with Concurrent Mode - mobx-lite does something similar. the UNSAFE_xxx handler is unsafe for concurrent mode.

how slow -- it's hard for me to say exactly right now, but the problem is that did mount is called after the changes are committed. So it has to go through DOM/native code before the second pass can occur -- even if changes are available synchronously. I guess that's the main bottleneck.

I will do some research on this issue

**Comment by radex** [comment]
@kokusGr Fixed your permissions, but since you're good to merge, I'm mergig

**Pull request merged**